jena-arq/src/main/java/org/apache/jena/query/Query.java

```diff
@@ -98,7 +98,7 @@ public class Query extends Prologue implements Cloneable, Printable
     public static final int ORDER_UNKNOW              = -3 ;
 
     // VALUES trailing clause
-    protected TableData valuesDataBlock = null ;
+    protected TableN valuesDataBlock = null ;
```

Comment:
Should the valuesDataBlock be mutable or not? One test case written by me some years ago assumed it to be mutable - but the implementation of TableData suggests to be immutable as the addBinding method is overridden to reject updates.

As I see it: TableN -> mutable, TableData -> immutable - much like guava's List and ImmutableList.

Reply:
Ok, so I'll change back to TableData and update the test case to expect failure when trying to modify the rows of a queries valuesDataBlock directly.

Reply:
Query has been reverted and the test case has been updated.